prow/gerrit/adapter/adapter.go

```diff
@@ -244,7 +244,7 @@ func (c *Controller) ProcessChange(instance string, change client.ChangeInfo) er
 		postsubmits := c.ca.Config().Postsubmits[cloneURI.String()]
 		postsubmits = append(postsubmits, c.ca.Config().Postsubmits[cloneURI.Host+"/"+cloneURI.Path]...)
 		for _, postsubmit := range postsubmits {
-			if postsubmit.RunsAgainstChanges(changedFiles) {
+			if postsubmit.RunIfChanged == "" || postsubmit.RunsAgainstChanges(changedFiles) {
```

Comment:
can't we encapsulate this into a function somewhere? We could make changedFiles be a function that returns the changed files (in case we don't need to look at them, aka most of the time).

Its hard to get this right.

Reply:
I thought about encapsulating `{pre,post}submit.ShouldRun()` but Gerrit seems to have entirely separate logic right now. I can try to move this in, instead.

Reply:
IMO gerrit should have the same logic. Worth breaking something (we control when deployments to all known gerrit instances happen)